napari/utils/events/containers/_evented_list.py

```diff
@@ -218,27 +287,38 @@ def move_multiple(
             else:
                 raise TypeError("Can only move integer or slice indices")
 
-        # remove duplicates while maintaing order, python 3.7+
         to_move = list(dict.fromkeys(to_move))
 
         if dest_index < 0:
             dest_index += len(self) + 1
-        dest_index -= len([i for i in to_move if i < dest_index])
-
-        self.events.moving(index=to_move, new_index=dest_index)
-        items = [self[i] for i in to_move]
-        for i in sorted(to_move, reverse=True):
-            self._list.pop(i)
-        for item in items[::-1]:
-            self._list.insert(dest_index, item)
-        self.events.moved(index=to_move, new_index=dest_index, value=items)
-        self.events.reordered(value=self)
-        return len(to_move)
+
+        d_inc = 0
+        popped: List[int] = []
+        for i, src in enumerate(to_move):
+            if src != dest_index:
+                # we need to decrement the src_i by 1 for each time we have
+                # previously pulled items out from in front of the src_i
+                src -= sum(map(_le(src), popped))
+                # if source is past the insertion point, increment src for each
+                # previous insertion
+                if src >= dest_index:
+                    src += i
+                yield src, dest_index + d_inc
+
+            popped.append(src)
+            # if the item moved up, icrement the destination index
+            if dest_index <= src:
+                d_inc += 1
 
     def reverse(self) -> None:
         """Reverse list *IN PLACE*."""
         # reimplementing this method to emit a change event
-        # If this method were removed, .reverse() would still be availalbe,
+        # If this method were removed, .reverse() would still be available,
         # it would just emit a "changed" event for each moved index in the list
         self._list.reverse()
         self.events.reordered(value=self)
+
+
+def _le(y):
+    """create a new function that accepts a single value x, returns x < y."""
```

Comment:
```suggestion
    """create a new function that accepts a single value x, returns x <= y."""
```

Not sure really worth it is the `map(_le(src), popped)` is replaced by a list comprehension, which may be more readable.

Reply:
oof... absolutely agree :)  replaced